packages/core/src/odata/v4/payload-value-converter.ts

```diff
@@ -107,9 +107,22 @@ function durationToEdmDuration(value: moment.Duration): string {
 }
 
 function timeToEdmTimeOfDay(value: Time): string {
-  return `${value.hours}:${value.minutes}:${value.seconds}`;
+  const hoursPadded = value.hours.toString(10).padStart(2, '0');
+  const minutesPadded = value.minutes.toString(10).padStart(2, '0');
+  const secondsPadded = padSeconds(value.seconds);
+  return `${hoursPadded}:${minutesPadded}:${secondsPadded}`;
 }
 
+function padSeconds(seconds: number): string {
+  const secondsString = seconds.toString(10);
+  if (!secondsString.includes('.')) {
+    return secondsString.padStart(2, '0');
+  }
+  const secondsStringParts: string[] = secondsString.split('.');
+  return [secondsStringParts[0].padStart(2, '0'), secondsStringParts[1]].join(
+    '.'
+  );
+}
```

Comment:
Actually you could use this function for all parts of the timeofday but even if you don't like that take a look below:
```suggestion
function padTimeComponent(timeComponent: number): string {
  const [wholeNumber, fractionalNumber] = timeComponent.toString().split('.');
  return [wholeNumber.padStart(2, '0'), fractionalNumber].join('.');
}
```